### PR TITLE
Add blockID to iterator error msg to aid debugging

### PR DESCRIPTION
**Pull request by annanay25**
Signed-off-by: Annanay <annanayagarwal@gmail.com>

### CODE REVIEW

File: tempodb/encoding/iterator_paged.go [hunk 1]
```diff
@@ -99,7 +102,7 @@ func (i *pagedIterator) Next(ctx context.Context) (common.ID, []byte, error) {
 	// attempt to get next object from objects
 	i.activePage, id, object, err = i.objectRW.UnmarshalAndAdvanceBuffer(i.activePage)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "error iterating through object in backend")
+		return nil, nil, fmt.Errorf("error iterating through object in backend, blockID: %s, err: %v", i.meta.BlockID.String(), err)
```
[the remark this answers is not in the document]
Comment (annanay25) [reply tempodb/encoding/iterator_paged.go]: 👍  good call.

File: tempodb/encoding/iterator_paged.go [hunk 2]
```diff
@@ -49,7 +52,7 @@ func (i *pagedIterator) Next(ctx context.Context) (common.ID, []byte, error) {
 	// dataReader returns pages in the raw format, so this works
 	i.activePage, id, object, err = i.objectRW.UnmarshalAndAdvanceBuffer(i.activePage)
 	if err != nil && err != io.EOF {
-		return nil, nil, errors.Wrap(err, "error iterating through object in backend")
+		return nil, nil, fmt.Errorf("error iterating through object in backend, blockID: %s, err: %v", i.meta.BlockID.String(), err)
```
[the remark this answers is not in the document]
Comment (annanay25) [reply tempodb/encoding/iterator_paged.go]: :facepalm:

File: tempodb/encoding/iterator_paged.go [hunk 3]
```diff
@@ -2,13 +2,15 @@ package encoding
...
+	"fmt"
 	"io"
 
+	"github.com/grafana/tempo/tempodb/backend"
 	"github.com/grafana/tempo/tempodb/encoding/common"
-	"github.com/pkg/errors"
 )
 
 type pagedIterator struct {
+	meta         *backend.BlockMeta
```
Comment (mdisibio) [inline tempodb/encoding/iterator_paged.go:13]: Would it be worth making this `blockID string` instead of the entire meta?

Comment (annanay25) [reply tempodb/encoding/iterator_paged.go:13]: No strong opinions, just felt its clearer to pass the meta around. Let me know if you feel otherwise

**Review by mapno — APPROVED** [review]
LGTM

**Pull request merged**